code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/contentcards/ContentCardUiProvider.kt

```diff
@@ -0,0 +1,181 @@
+import com.adobe.marketing.mobile.Messaging
+import com.adobe.marketing.mobile.aepcomposeui.AepUI
+import com.adobe.marketing.mobile.aepcomposeui.contentprovider.AepUIContentProvider
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.AepUITemplate
+import com.adobe.marketing.mobile.messaging.MessagingConstants
+import com.adobe.marketing.mobile.messaging.Proposition
+import com.adobe.marketing.mobile.messaging.SchemaType
+import com.adobe.marketing.mobile.messaging.Surface
+import com.adobe.marketing.mobile.services.Log
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+/**
+ * ContentCardUiProvider is responsible for managing the content flow of [AepUI] elements
+ * for a given surface string. It fetches, updates, and refreshes content asynchronously.
+ *
+ * @property surfaceString The string representing the surface for which content is fetched.
+ */
+
+class ContentCardUiProvider(val surfaceString: String) : AepUIContentProvider {
+    companion object {
+        private const val SELF_TAG: String = "ContentCardUiProvider"
+    }
+
+    private val _contentFlow = MutableStateFlow<List<AepUI<*, *>>>(emptyList())
+    private val contentFlow: StateFlow<List<AepUI<*, *>>> = _contentFlow
+
+    /**
+     * Retrieves the content as a Flow of a list of [AepUI] elements.
+     * The content is fetched asynchronously for the provided surface string.
+     *
+     * @return A Flow of a list of [AepUI] elements.
+     */
+    override suspend fun getContent(): Flow<List<AepUI<*, *>>> {
+        CoroutineScope(Dispatchers.IO).launch {
+            val surface = Surface(surfaceString)
+            val surfaceList = mutableListOf<Surface>()
+            surfaceList.add(surface)
+            getAepUIForSurfaceList(surfaceList) {
+                it.onSuccess { templateList ->
+                    _contentFlow.value = templateList.map { provideAepUIfromAepUITemplate(it) }
+                }
+                it.onFailure { error ->
+                    Log.error(
+                        MessagingConstants.LOG_TAG, SELF_TAG,
+                        "Failed to get content: ${error.message}"
+                    )
+                    _contentFlow.value = emptyList()
+                }
+            }
+        }
+        return contentFlow
+    }
+
+    /**
+     * Retrieves the list of [AepUITemplate] for the provided surfaces.
+     * Updates the propositions and retrieves templates based on those propositions.
+     *
+     * @param surfaceList The list of surfaces for which to retrieve the templates.
+     * @param completion The callback to handle the result of the template retrieval.
+     */
+
+    private suspend fun getAepUIForSurfaceList(
+        surfaceList: MutableList<Surface>,
+        completion: (Result<List<AepUITemplate>>) -> Unit
```

Comment:
Can we keep method signatures consistent here? One returns a `Flow` another uses a callback. Can we please stick to using Flow and make this a suspend function?

Also why do we need both `getAepUIForSurfaceList` and `provideAepUIfromAepUITemplate`? Aren't they doing the same thing? Can we keep one of them?

Reply:
Removed

Reply:
This is not a used facing method, but an internal method to be used by getContent() method which will be called by getAepUI()